setuptools_rust/build.py

```diff
@@ -95,14 +96,18 @@ def build_extension(self, ext):
         # Automatic target detection can be overridden via the CARGO_BUILD_TARGET
         # environment variable.
         # TODO: include --target for all platforms so env vars can't break the build
-        target_triple = None
         target_args = []
-        if os.getenv("CARGO_BUILD_TARGET"):
-            target_triple = os.environ["CARGO_BUILD_TARGET"]
-        elif self.plat_name == "win32":
-            target_triple = "i686-pc-windows-msvc"
-        elif self.plat_name == "win-amd64":
-            target_triple = "x86_64-pc-windows-msvc"
+        if target_triple is None:
+            if os.getenv("CARGO_BUILD_TARGET"):
+                target_triple = os.environ["CARGO_BUILD_TARGET"]
+            elif self.plat_name == "win32":
+                target_triple = "i686-pc-windows-msvc"
+            elif self.plat_name == "win-amd64":
+                target_triple = "x86_64-pc-windows-msvc"
+            elif self.plat_name.startswith("macosx-"):
+                if platform.machine() == "arm64" and platform.processor() == "i386":
+                    # arm64 macOS targeting x86_64
+                    target_triple = "x86_64-apple-darwin"
```

Comment:
Should this logic maybe be moved into a `get_target_triple()` function? It's getting quite long.